apps/api/src/app/subscribers/usecases/chat-oauth/chat-oauth.usecase.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+
+import { ChannelTypeEnum } from '@novu/stateless';
+import { IntegrationEntity, IntegrationRepository } from '@novu/dal';
+
+import { ChatOauthCommand } from './chat-oauth.command';
+
+@Injectable()
+export class ChatOauth {
+  readonly SLACK_OAUTH_URL = 'https://slack.com/oauth/v2/authorize?';
+
+  constructor(private integrationRepository: IntegrationRepository) {}
+  async execute(command: ChatOauthCommand): Promise<string> {
+    const clientId = await this.getClientId(command);
+
+    return this.getOAuthUrl(command.subscriberId, command.environmentId, clientId);
+  }
+
+  private getOAuthUrl(subscriberId: string, environmentId: string, clientId: string): string {
+    const redirectUri = `${process.env.API_ROOT_URL}/v1/subscribers/${subscriberId}/credentials/slack/${environmentId}/callback`;
+
+    return `${this.SLACK_OAUTH_URL}client_id=${clientId}&scope=incoming-webhook&user_scope=&redirect_uri=${redirectUri}`;
+  }
+
+  private async getClientId(command: ChatOauthCommand): Promise<string> {
+    const query: Partial<IntegrationEntity> & { _environmentId: string } = {
+      _environmentId: command.environmentId,
+      channel: ChannelTypeEnum.CHAT,
+      providerId: command.providerId,
+    };
+
+    const integration = await this.integrationRepository.findOne(query);
```

Comment:
@djabarovgeorge what would happen if here he finds a different chat integration and not slack? Also, does it matter here if its active or not?

Reply:
I think it will be ok, Although we only support Slack at the moment this controller/usecase should be provider agnostic meaning we could add support to other providers as well.
Regarding the second question, you raised a good point, this implementation does not care if the integration is active or not, leaving the responsibility to the user.

Reply:
But if the query findOne returns the discord integration?

Reply:
sorry sorry